src/programs.c

```diff
@@ -330,36 +332,83 @@ static __always_inline syscall_pattern_type_t ptrace_syscall_pattern(u32 request
-    /* if "loaded" doesn't exist in the map, we get NULL back and won't read from offsets               \
-     * when offsets are loaded into the offsets map, "loaded" should be given any value                 \
-     */                                                                                                 \
-    u64 loaded = CRC_LOADED;                                                                            \
-    loaded = (u64)bpf_map_lookup_elem(&offsets, &loaded); /* squeezing out as much stack as possible */ \
-    /* since we're using offsets to read from the structs, we don't need to bother with                 \
-     * understanding their structure                                                                    \
-     */                                                                                                 \
-    void *ts = (void *)bpf_get_current_task();                                                          \
-    void *pts = NULL;                                                                                   \
-    if (ts && loaded)                                                                                   \
-    {                                                                                                   \
-        u64 real_parent_offset = CRC_TASK_STRUCT_REAL_PARENT;                                           \
-        u64 ppid_offset = CRC_TASK_STRUCT_PID;                                                          \
-        u64 luid_offset = CRC_TASK_STRUCT_LOGINUID;                                                     \
-        real_parent_offset = (u64)bpf_map_lookup_elem(&offsets, &real_parent_offset);                   \
-        ppid_offset = (u64)bpf_map_lookup_elem(&offsets, &ppid_offset);                                 \
-        luid_offset = (u64)bpf_map_lookup_elem(&offsets, &luid_offset);                                 \
-        if (real_parent_offset && ppid_offset && luid_offset)                                           \
-        {                                                                                               \
-            bpf_probe_read(&pts, sizeof(pts), ts + *(u32 *)real_parent_offset);                         \
-            if (pts)                                                                                    \
-            {                                                                                           \
-                bpf_probe_read(&ppid, sizeof(ppid), pts + *(u32 *)ppid_offset);                         \
-            }                                                                                           \
-            bpf_probe_read(&luid, sizeof(luid), ts + *(u32 *)luid_offset);                              \
-        }                                                                                               \
+#define FILL_TELEMETRY_SYSCALL_RET(E, SP) \
+
+#define SET_OFFSET(CRC) \
```

Comment:
That looks like it should work, I'll try it out. My only concern is inflating the instruction count since the `4_11` probe is very close to the `4096` limit.